### PR TITLE
Prevent app crashes when pinning a Catalog Entity to Hotbar

### LINKED ISSUE #2813: Pin Catatlog Entity to Hotbar crash the app on restart
**Describe the bug**
Pin a Catalog Entity to Hotbar causes crashes

**To Reproduce**
Pin a Catalog Entity to Hotbar

Screen Shot 2021-05-19 at 2 52 49 PM

Confirm it's in Hotbar
Screen Shot 2021-05-19 at 2 54 49 PM

Restart Lens.

See this
Screen Shot 2021-05-19 at 2 57 40 PM

In console
Screen Shot 2021-05-19 at 2 58 41 PM

Above related to 
https://github.com/lensapp/lens/blob/master/src/renderer/components/hotbar/hotbar-menu.tsx#L55

If I only fix `hotbar-menu.tsx`

I will see this error in console instead

Screen Shot 2021-05-19 at 3 11 26 PM

This need fix too
https://github.com/lensapp/lens/blob/master/src/renderer/api/catalog-entity-registry.ts#L68

**Expected behavior**

No crashes

### PR DESCRIPTION
**Pull request by chenhunghan**
Fix https://github.com/lensapp/lens/issues/2813

But I am not sure it's a proper fix. Please review/confirm.

### CODE REVIEW

File: src/renderer/api/catalog-entity-registry.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class CatalogEntityRegistry {
 
   getItemsForCategory<T extends CatalogEntity>(category: CatalogCategory): T[] {
     const supportedVersions = category.spec.versions.map((v) => `${category.spec.group}/${v.name}`);
-    const items = this._items.filter((item) => supportedVersions.includes(item.apiVersion) && item.kind === category.spec.names.kind);
+    const items = this._items.filter((item) => supportedVersions.includes(item?.apiVersion) && item.kind === category.spec.names.kind);
```
Comment (Nokel81) [inline src/renderer/api/catalog-entity-registry.ts]: I think it would be better if `this._items` filtered out null or undefined values.

Comment (chenhunghan) [reply src/renderer/api/catalog-entity-registry.ts]: thanks, good point

Comment (chenhunghan) [reply src/renderer/api/catalog-entity-registry.ts]: Fixed in https://github.com/lensapp/lens/pull/2814/commits/b0cff92ca804ee91bf8aaa784ab2571fba2a7967

Comment (Nokel81) [reply src/renderer/api/catalog-entity-registry.ts]: I meant on line 41. Namely keep the "database" clean.

Comment (chenhunghan) [reply src/renderer/api/catalog-entity-registry.ts]: ah, ok, sorry, fixed https://github.com/lensapp/lens/pull/2814/commits/0da59e998ed2fd3cc38251f6a6f5155c571ca050

File: src/renderer/components/hotbar/hotbar-menu.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ export class HotbarMenu extends React.Component<Props> {
       return null;
     }
 
-    return item ? catalogEntityRegistry.items.find((entity) => entity.metadata.uid === item.entity.uid) : null;
+    return item ? catalogEntityRegistry.items.find((entity) => entity?.metadata.uid === item.entity.uid) : null;
```
Comment (Nokel81) [inline src/renderer/components/hotbar/hotbar-menu.tsx]: Is this still needed then?

Comment (chenhunghan) [reply src/renderer/components/hotbar/hotbar-menu.tsx]: no, removed https://github.com/lensapp/lens/pull/2814/commits/79740943be47f4414e715b1876c1992591184b94

**Review by Nokel81 — APPROVED** [review]

**Review by jakolehm — APPROVED** [review]

**Comment by Nokel81** [comment]
@jakolehm Apparently this causes other problems, so I opened https://github.com/lensapp/lens/pull/2816 as an actual fix.

**Comment by chenhunghan** [comment]
This should be closed because #2816 is the "fix"

**Pull request closed without merging**